src/Base/Exception.h

```diff
@@ -33,6 +33,13 @@
 #include "FileInfo.h"
 #include "BaseClass.h"
 
+/* MACROS FOR THROWING EXCEPTIONS */
+
+#define THROW(exception) throw Base::exception(__FILE__,__LINE__,__PRETTY_FUNCTION__);
+#define THROWM(exception, message) throw Base::exception(message,__FILE__,__LINE__,__PRETTY_FUNCTION__);

```

Comment:
@abdullahtahiriyo - I didn't mean to prompt so much discussion, sorry.

Basically, I don't see a bug report that identifies the line where an exception was thrown, as much more valuable than just one that has some chunk of text that can be grepped for.

So, to me, this seems like a reasonably large amount of maintenance effort to support a thing that I don't really see the point of.